lib/spack/spack/build_systems/cmake.py

```diff
@@ -314,7 +314,8 @@ def build_directory(self):
 
         :return: directory where to build the package
         """
-        return os.path.join(self.stage.path, 'spack-build')
+        dirname = 'spack-build=%s' % self.spec.dag_hash(7)
```

Comment:
I didn't notice the `=` sign on the directory name. Can we maybe use an `-` instead:
```suggestion
        dirname = 'spack-build-%s' % self.spec.dag_hash(7)
```
?